docs/concepts/submitting-tasks-jobs-to-functions.md

```diff
@@ -17,8 +17,13 @@ Functions can host multiple methods (handlers). You can set the default handler
  need to specify which handler you intend to call in the run command. 
 
 You can pass `parameters` (arguments) or data `inputs` (such as datasets, feature-vectors, models, or files) to the functions through the `run` method.
-- Inside the function you can access the parameters/inputs by simply adding them as parameters to the function, or you can get them from the context object (using `get_param()` and ` get_input()`).
-- Various data objects (files, tables, models, etc.) are passed to the function as data item objects. You can pass data objects using the 
+
+* **Parameters** (`params`) are meant for basic python objects that are serializable, meaning they can be parsed from text without special handling. So, passing `int`, `float`, `str` and `dict`, `list` are all possible using `params`. MLRun takes the parameter and assigns it to the relevant handler parameter by name.
+* **Inputs** are meant for getting local or remote data as `mlrun.DataItem`. Using data items reduces the glue logic required for getting data from remote files (like files located in S3) and local files, as well as logged artifacts. It has many methods like `local` (download the data item's file to a local temp directory) and `as_df` (parse the data to a `pd.DataFrame`). You can read more [here](../store/data-items.html).
+
+Inside the function you can access the parameters/inputs by simply adding them as parameters to the function, or you can get them from the context object (using `get_param()` and ` get_input()`).
```

Comment:
this is True but we would rather keep users away from MLRun's context. It was the correct and only way to get the parameters and inputs but now its not and I think it is better to promote in the concepts section the bet practices and this can be explained at "what is MLRun's context" or something like that. 

What I would add is that when a type hint is available for an argument MLRun will automatically parse the `DataItem` to the hinted type (when the hinted type is supported).

Reply:
@guy1992l  Do you mean to move this sentence:
Inside the function you can access the parameters/inputs by simply adding them as parameters to the function, or you can get them from the context object (using `get_param()` and ` get_input()`).
to https://docs.mlrun.org/en/latest/concepts/mlrun-execution-context.html ?

Reply:
yes :)

Reply:
done